docs/Trimming.md

```diff
@@ -0,0 +1,50 @@
+# Guide to trimming ASP.NET Core
+
+This guide discusses the steps required to enable, annotate, and verify trimming in ASP.NET Core assemblies. An assembly that supports trimming knows all the types and members it needs at runtime. Unused code is removed when an app is published, reducing app size. Trimming is also a precursor to supporting AOT, which requires knowledge of the APIs used by an app on build.
+
+[Trim self-contained deployments and executables](https://docs.microsoft.com/dotnet/core/deploying/trimming/trim-self-contained) has general documentation for trimming. It includes:
+
+* How to output trim warnings. Typically done by publishing an app.
+* How to react to trim warnings. Attributes can be placed on APIs to provide type info, suppress warnings, or verify that an API is incompatible with trimming.
+* How to trim libraries.
+
+## Assembly trimming order
+
+Trim assemblies from the bottom up. Order is important because annotating an assembly impacts its dependents and their annotations. Annotating from the bottom up reduces churn.
+
+For example, `Microsoft.AspNetCore.Http` depends on `Microsoft.AspNetCore.Http.Abstractions` so `Microsoft.AspNetCore.Http.Abstractions` should be annotated first.
+
+## Trim an ASP.NET Core assembly
+
+The first step to trimming an ASP.NET Core assembly is adding it to `LinkabilityChecker`. `LinkabilityChecker` is a tool in the ASP.NET Core repo that runs ILLink on its referenced assemblies and outputs trim warnings.
+
+1. Update the project file to enable trimming `<Trimmable>true</Trimmable>`.
```

Comment:
Does this do something more than `IsTrimmable` for ASP.NET?

Reply:
It also drives a list of projects that are passed to LinkabilityChecker tool in this repo

See https://github.com/dotnet/aspnetcore/pull/40342 and https://github.com/dotnet/aspnetcore/pull/40394

I'm not sure of the exact backstory of why we use a different property to enable trimming.

Reply:
> I'm not sure of the exact backstory of why we use a different property to enable trimming.

I believe this was because we started trimming before the IsTrimmable MSBuild property was created. We had a similar one-off property in dotnet/runtime and later moved to the official property in https://github.com/dotnet/runtime/pull/65870.

Reply:
That's probably it. Issue for moving to new property: https://github.com/dotnet/aspnetcore/issues/41016